packages/web-client/app/components/card-pay/issue-prepaid-card-workflow/index.ts

```diff
@@ -57,6 +61,22 @@ class IssuePrepaidCardWorkflow extends Workflow {
     new Milestone({
       title: 'Customize layout',
       postables: [
+        new WorkflowCheck({
+          author: cardbot,
+          check: async () => {
+            let layer2Network = this.owner.lookup(
+              'service:layer2-network'
+            ) as Layer2Network;
+
+            // TODO: replace this with usage of ExchangeRate.convertToSpend
+            let daiMinValue = `${Math.min(...faceValueOptions) / 100}`;
+
+            return !!layer2Network.defaultTokenBalance?.gte(
+              new BN(toWei(daiMinValue))
+            );
+          },
+          failureReason: 'INSUFFICIENT_FUNDS',
```

Comment:
I’m not sure what the interface for this would be, but it seems like there’s a gap here for some dynamicism in rendering the failure reason, because I think it would be helpful to be able to specify not just that the funds are insufficient, but also what amount would be sufficient. Maybe this could be addressed in future work, but I think it’s likely we’ll need to inject arbitrary values at runtime vs build time. But for now maybe the minimum can be included in the insufficient funds message?

Reply:
also, this looks like a one-time check, I wonder if there might be a need for a polling- or event-based check, like what if the balance changes from other operations and then is below the minimum?

Reply:
There are event based checks, however those are not done declaratively but via the `Listener` component and actions in workflows.